routemaster/models/queue.rb

```diff
@@ -163,11 +163,25 @@ module ClassMethods
         include Mixins::Redis
         include Enumerable
 
+        # Names of all used queues
+        NAMES = %w[ aux main ]
+
+        # Accessor for allowed named queues
+        def [](name)
+          raise ArgumentError, "queue name '#{name}' is not valid" unless NAMES.include?(name)
+          new(name: name)
+        end
+
         def each
-          _redis.scan_each(match: 'jobs:index:*') do |k|
-            yield new(name: k.sub(/^jobs:index:/, ''))
+          NAMES.each do |n|
+            yield new(name: n)
```

Comment:
Kind of related to my comment above, if the well known queues were constants/singletons you could save yourself a bunch of object allocations here. Not a huge deal, but hey if we're optimising stuff 😉